### PR TITLE
[10.0][ADD][project_task_dependency] Task dependencies on tasks

### PR DESCRIPTION
**Pull request by tarteo**
Enables the user to define dependencies (other tasks) of a task.
e.g. task 'building the roof' is dependent on task 'building walls and pillars'

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please put the module name in singular: `project_task_dependency` according guidelines https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules

**Review by rafaelbn — APPROVED** [review]
Functional test

Good to know about https://github.com/OCA/project/tree/8.0/project_recalculate module which is very interesting to calculate dates depending in dependencies.

**Comment by levkar** [comment]
It might be better to have two many2many fields (predecessor_ids and successor_ids) which uses same table instead of task_ids.

**Comment by susport** [comment]
You could validate not save tasks with the same name, when a task depends on the other.

**Comment by bistaray** [comment]
What is the best way to review this?  I see no runbot link, and pulling the v10 branch does not provide this module.  (Sorry - am new with GIT)

**Comment by jcdrubay** [comment]
You can pull the code from here: https://github.com/tarteo/project/tree/100_project_task_dependencies

**Comment by pedrobaeza** [comment]
@bistaray, runbot is available clicking on "Show all checks"

**Comment by bistaray** [comment]
@pedrobaeza - thanks - I see this now.  admin / admin isn't working.  How do I discover the credentials?

**Comment by pedrobaeza** [comment]
@bistaray anyone can change the password. I have launched a rebuilt of the instance to have a fresh one with the default password.

File: project_task_dependencies/models/project_task.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo import models, fields
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    task_ids = fields.Many2many(
+        comodel_name='project.task', relation='project_task_project_task_rel',
+        column1='task_id', column2='dependent_task_id', string='Dependencies',
```
Comment (dreispt) [inline project_task_dependencies/models/project_task.py]: This is not clear to me: the are the Tasks I depend on, or the Tasks the depend on me?
The field name (and string label) should make this clear.

**Review by dreispt — CHANGES_REQUESTED** [review]
There is also a pending request to change the module name.

File: project_task_dependencies/models/project_task.py (hunk 1 above)
Comment (bistaray) [reply project_task_dependencies/models/project_task.py]: "Dependencies" is clear to me, and I'd assert to native English speakers.  See http://www.dictionary.com/browse/dependencies  and https://blogs.msdn.microsoft.com/project/2008/07/28/back-to-basics-understanding-task-dependencies/ 

Another option would be 'Predecessors' (things that come before me)

**Comment by pedrobaeza** [comment]
I remind you that the module names should be in singular form according guidelines.

**Comment by bistaray** [comment]
Using runbot I was able to add task dependencies to another task.  This module only offers the basic data structure.

Possibly users may wish a recursion check - to ensure no circular dependencies.

I will want a way to visualize dependencies across a project - so would need a field 'parent_id' or equivalent - so I can list tasks in a treeview (and group by parent perhaps?) showing the parent task for each task.  @levkar has suggested an improvement that may make visualization easier also.

As it stands it works and is respects the design of Odoo, IMHO.

**Comment by tarteo** [comment]
I've fixed the name and created a mechanism to prevent circular dependencies.

**Review by dreispt — APPROVED** [review]

**Comment by tarteo** [comment]
@dreispt Can you remove the needs fixing tag, please?

**Pull request merged**